### PR TITLE
Improve mention suggestions list

### LINKED ISSUE #12350: Improve mention suggestions list: remove ENS + restrict to group (community members)
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, in mention suggestion list, I want to see only users who are participating in the current chat.

### Description
Based on [discussion](https://discord.com/channels/624205794384281629/743483790898233374/867685518849605662) expected behavior for the mention suggestion list should be changed.
Now in the suggestion list, users can see all resolved ENS names + contacts even if they are not participating in the current chat, so they could not be notified about the mention.

*Summary*: ...

### Acceptance Criteria

- [ ] Exclude ENS names from the mention suggestion list if they are not participating in the current chat (group, community, 1-1)
- [ ] Exclude contacts from the mention suggestion list if they are not participating in the current chat (group, community, 1-1)

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #12350 

### Summary

Remove unnecessary suggestions from mentions suggestions list

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

### Steps to test

- Open Status
- Open group chat or 1-on-1 chat
- Write "@" to see mentions suggestions
- Verify that only contacts in the chat appear in the suggestions list

status: ready

### CODE REVIEW

**Review by bitgamma — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @briansztamfater! Could you rebase this against the latest develop, please?
Thanks

**Comment by briansztamfater** [comment]
> Hey @briansztamfater! Could you rebase this against the latest develop, please?
> Thanks

Never saw this, sorry! Rebase done :)

**Comment by qoqobolo** [comment]
@briansztamfater thanks for your work!
There are some issues to address.

### ISSUE 1: Nicknames are shown in the mention suggestions list only after relogin

**_Reproduction:_**
1. Join a public
2. Open profile of some user
3. Set a nickname
4. Type `@` to check suggestions
5. Relogin and repeat step 4

**Expected result:** nickname is displayed in the mention suggestions list right after it is set/changed
**Actual result:** nickname is displayed only after relogin

https://user-images.githubusercontent.com/67952253/134191418-eee1dcf4-3b58-485c-862d-dfe0cb48790f.mp4

### ISSUE 2: ENS names are not resolved in the mention suggestions list in Android and are not resolved at all on iOS

On iOS can only see my own ENS both in the mention suggestions list and the chat history.

**_Reproduction:_**
1. Join a public chat with ENS user (start a 1-1 chat or create a group)
2. Type `@` and check the suggestions list

**Expected result:** 
- Android: ENS names are displayed in the mention suggestions list
- iOS: can see ENS names in the mention suggestions list, chat history, header of 1-1 chat, etc.

**Actual result:** ENS names are only displayed after sending a message with a mention in the chat history (on Android)

https://user-images.githubusercontent.com/67952253/134192396-42f477fb-eb87-4697-94d8-3b17b862a67c.mp4

### ISSUE 3/Q: Is the behavior expected to display a mention suggestion for a user in the suggestions list only after he has sent something to the chat?

In other words, until the user sends something to the chat, his name will not be in the list. The behavior is relevant for any type of chat. 

**_Reproduction:_**
1. UserA: create a community and invite UserB
2. UserB: join the community
3. UserA: type `@`

**Actual result:** no suggestions for UserB in the list, although he has already joined the community

**Comment by flexsurfer** [comment]
still relevant ?

**Comment by briansztamfater** [comment]
@flexsurfer yeh I will resume it ASAP, currently focused on other issue

**Comment by flexsurfer** [comment]
no rush, just looked which PRs can be closed

**Comment by briansztamfater** [comment]
@qoqobolo sorry for the long wait, branch is now rebased and issues should be addressed

**Comment by qoqobolo** [comment]
Thanks @briansztamfater!

### ISSUE 4: Names of users who left the group chat continue to be displayed in the mention suggestions list

**Steps:**
1. UserA creates a group chat with UserB
2. UserB joins it and then leaves
3. UserA checks the mentions list

**Expected result:** UserB name is no longer in the list
**Actual result:** the name is still there

WDYT? Is it possible to fix it in this PR?

**Comment by briansztamfater** [comment]
Good catch @qoqobolo, I will look at it, ideally should be fixed in this PR

**Comment by briansztamfater** [comment]
@qoqobolo Issue was (hopefully) fixed, let me know if anything arises during testing :)

**Comment by qoqobolo** [comment]
Thanks @briansztamfater! LGTM!
PR is ready can be merged.
